dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/mapper/ProgramStageInstanceMapper.java

```diff
@@ -108,6 +108,9 @@ else if ( importOptions.getIdSchemes().getProgramStageIdScheme().equals( UID ) )
 
         psi.setStoredBy( event.getStoredBy() );
         psi.setCompletedBy( event.getCompletedBy() );
+        
+        // STATUS
+        psi.setStatus( fromInt( event.getStatus().getValue() ) );
```

Comment:
Completed date only gets updated if "psi.isCompleted()" returns true, which in turn checks the status. So we need to update the status first before proceeding to update completed date.

Reply:
got it